nntrainer/utils/ini_wrapper.h

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file ini_wrapper.h
+ * @date 08 April 2021
+ * @brief NNTrainer Ini Wrapper helps to save ini
+ * @note this is to be used with ini_interpreter
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#include <iostream>
+#include <map>
+#include <string>
+
+#ifndef __INI_WRAPPER_H__
+#define __INI_WRAPPER_H__
+
+namespace nntrainer {
+
+/**
+ * @brief IniSection class that maps to one ini section
+ *
+ */
+class IniSection {
+
+public:
+  /**
+   * @brief Construct a new Ini Section object
+   *
+   * @param name section name
+   */
+  IniSection(const std::string &name);
+
+  /**
+   * @brief Construct a new Ini Section object
+   *
+   * @param section_name section name
+   * @param entry_str entry representing string (separated by `|`)
+   */
+  IniSection(const std::string &section_name, const std::string &entry_str);
+
+  /**
+   * @brief Copy Construct a new Ini Section object
+   * @note this function copies entry from @a from and overwrite entry and
+   * section_name
+   *
+   * @param from Ini Section to copy from
+   * @param section_name section name to override, if empty, section name is not
+   * updated
+   * @param entry_str entry string to override the given section name
+   */
+  IniSection(IniSection &from, const std::string &section_name,
+             const std::string &entry_str);
+
+  /**
+   * @brief Construct a new Ini Section object
+   *
+   * @param from Ini Section to copy from
+   * @param entry_str entry string to override the given section name
+   */
+  IniSection(IniSection &from, const std::string &entry_str) :
+    IniSection(from, "", entry_str) {}
+
+  IniSection() = default;
```

Comment:
is this needed?

Reply:
Yes, trivial default ctor is needed for the containers (eg) `std::vector<IniSection>`)

Reply:
ok.